File: 347. Top K Frequent Elements/attempt_2.cpp
#include<iostream>
#include<vector>
#include<queue>
#include<unordered_map>

using namespace std;
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        for(int x : nums)
            freq[x]++;
        priority_queue<
            pair<int,int>,
            vector<pair<int,int>>,
            greater<pair<int,int>>
        > pq;

        for(auto& [num, f] : freq) {
            pq.push({f, num});
            if(pq.size() > k)
                pq.pop();
        }

        vector<int> result;
        while(!pq.empty()) {
            result.push_back(pq.top().second);
            pq.pop();
        }
        return result;
    }
};
int main() {
    Solution sol;
    vector<int> nums = {1,2,1,2,1,2,3,1,3,2};
    for(auto x : sol.topKFrequent(nums, k)) {
        cout << x << " ";
    }
}

File: 41. First Missing Positive/attempt2.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)
        i = 0
        while i < n:
            curr = nums[i]
            nums[i] = 0
            while 1 <= curr <= n:
                target = curr - 1
                if nums[target] == curr:
                    break
                curr, nums[target] = nums[target], curr
            i += 1
        for i in range(n):
            if nums[i] != i + 1:
                return i + 1
        return n + 1

sol = Solution()
nums = [3,4,-1,-2,1,5,16,0,2,0]
nums = [3,4,-1, 1]
print(sol.firstMissingPositive(nums))